maths/area.py

```diff
@@ -82,34 +78,61 @@ def area_square(side_length: float) -> float:
     return side_length ** 2
 
 
-def area_triangle(base: float, height: float) -> float:
+def area_isosceles_triangle(base: float, height: float) -> float:
     """
-    Calculate the area of a triangle
-
-    >>> area_triangle(10, 10)
+    Calculate the area of an isosceles triangle
+    >>> area_isosceles_triangle(10, 10)
     50.0
-    >>> area_triangle(-1, -2)
+    >>> area_isosceles_triangle(-1, -2)
     Traceback (most recent call last):
         ...
-    ValueError: area_triangle() only accepts non-negative values
-    >>> area_triangle(1, -2)
+    ValueError: area_isosceles_triangle() only accepts non-negative values
+    >>> area_isosceles_triangle(1, -2)
+    Traceback (most recent call last):
+        ...
+    ValueError: area_isosceles_triangle() only accepts non-negative values
+    >>> area_isosceles_triangle(-1, 2)
+    Traceback (most recent call last):
+        ...
+    ValueError: area_isosceles_triangle() only accepts non-negative values
+    """
+    if base < 0 or height < 0:
+        raise ValueError("area_isosceles_triangle() only accepts non-negative values")
+    return (base * height) / 2
+
+
+def area_triangle(side1: float, side2: float, side3: float) -> float:

```

Comment:
This is finding the area of the triangle using a specific formula that requires the length of all three sides: [Heron's formula](https://en.wikipedia.org/wiki/Heron%27s_formula). So, please update the name accordingly.